teacher.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './person'
+
+class Teacher < Person
+  def initialize(specialization)
+    @specialization = specialization
+    super(age, name, parent_permission: parent_permission)
+  end
```

Comment:
- [ ] - [ ] As required for the project, kindly make sure you use the same approach like on the `Student` class. The `initialize` constructor method extends parent's constructor by adding `@specialization` and a `parameter` for it. This means  constructor method should also include all parameters of a `Person` + ``specialization``. Make sure you correct this.
```ruby
   def initialize(age, `specialization`, name = "Unknown", parent_permission = true)
   ...
```

Reply:
Okay. Thanks. I will walk through it.